src/mon/OSDMonitor.cc

```diff
@@ -1663,6 +1665,12 @@ bool OSDMonitor::check_failure(utime_t now, int target_osd, failure_info_t& fi)
     for (map<int,failure_reporter_t>::iterator p = fi.reporters.begin();
 	 p != fi.reporters.end();
 	 ++p) {
+      //get subtree reporter
+      map<string,string> reporter_loc = osdmap.crush->get_full_location(p->first);
+      int osd_subtree_reporter  = osdmap.crush->get_item_id(reporter_loc[reporter_subtree_level]);
+      assert(osd_subtree_reporter != 0);
```

Comment:
I don't think we should assert here.  The user could configure the level to be "host" but the crush map might have an osd that is not under a host (say it's attached directly to the root or a rack or something).  That's a misconfiguration, but we shouldn't crash.  If get_item_id fails here let's call the reporter "unknown"? And may print a warning to the mon log in that case